modules/password/password.go

```diff
@@ -101,7 +102,11 @@ func Generate(n int) (string, error) {
 			}
 			buffer[j] = validChars[rnd.Int64()]
 		}
-		if IsComplexEnough(string(buffer)) && string(buffer[0]) != " " && string(buffer[n-1]) != " " {
+		pwned, err := IsPwned(goContext.Background(), string(buffer))
```

Comment:
Don't use context.Background() it's not cancellable - if you can't get the request context always use the graceful.GetManager().ShutdownContext() 

(unless you're in a module that is creates it's own cancellable context from Background that is tied to Shutdown - see how the contexts in Queue are tied to the Shutdown context through Shutdown callbacks. It's done that way to prevent Queues depending too much on graceful.)